build-logic/jvm/src/main/kotlin/gradlebuild/jvm/extension/UnitTestAndCompileExtension.kt

```diff
@@ -16,47 +16,79 @@
 
 package gradlebuild.jvm.extension
 
-import org.gradle.api.Project
-import org.gradle.api.tasks.TaskContainer
-import org.gradle.api.tasks.compile.JavaCompile
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
 import org.gradle.kotlin.dsl.*
 
+/**
+ * An extension intended for configuring the manner in which a project is
+ * compiled and tested.
+ */
+abstract class UnitTestAndCompileExtension {
```

Comment:
Usage looks like:
```
gradlebuildJava {
    usesJdkInternals = true
    usedInWorkers()
}
```

I also think something like 
```
gradlebuildJava {
    usedInWorkers()
    workarounds {
        usesJdkInternals = true
    }
}
```

would be good, since it makes it very clear these flags are not something you should rely on for your project